src/Type/DynamicFunctionParameterOutTypeExtension.php

```diff
@@ -0,0 +1,32 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParameterReflection;
+
+/**
+ * This is the interface dynamic parameter out type extensions implement for functions.
+ *
+ * To register it in the configuration file use the `phpstan.dynamicFunctionReturnTypeExtension` service tag:
+ *
+ * ```
+ * services:
+ * 	-
+ *		class: App\PHPStan\MyExtension
+ *		tags:
+ *			- phpstan.dynamicFunctionReturnTypeExtension
+ * ```
+ *
+ * @api
+ */
+interface DynamicFunctionParameterOutTypeExtension
```

Comment:
We've recently started dropping the "dynamic" part of the name from an extension. See `FunctionParameterClosureTypeExtension` for example.

Since you're also applying it here inconsistently (you're not mentioning "dynamic" in the PR title, and also `LazyParameterOutTypeExtensionProvider` doesn't contain it), please drop it from all the names in this PR. Thanks.

Reply:
Fixed